src/jdk.jfr/share/classes/jdk/jfr/internal/EventInstrumentation.java

```diff
@@ -90,65 +89,86 @@ public FieldInfo(String fieldName, String fieldDescriptor, String internalClassN
         }
 
         public boolean isString() {
-            return STRING.getDescriptor().equals(fieldDescriptor);
+            return ASMToolkit.TYPE_STRING.getDescriptor().equals(fieldDescriptor);
```

Comment:
If this is to be called often it might be beneficial to cache the descriptor instead of recreating it each time.

Reply:
Turns out this method is not needed, so I removed it. I also changed the classes FieldInfo and SettingInfo into record classes and while at it, I also made EventInstrumentation (shallow) immutable so guardEventConfiguration is now passed in the constructor.